scripts/upstream-release-bot.py

```
@@ -902,23 +1333,104 @@ def main(argv: list[str]) -> int:
+        if not isinstance(package_state, dict):
+            package_state = {}
+            state_packages[update.package] = package_state
+        try:
+            validate_package_generated_paths(
                 repo_root=repo_root,
+                package=update.package,
                 staged_paths=staged_paths,
+            )
+        # Keep package validation failures isolated: restore staged writes,
+        # record the error, quarantine this package, and continue others.
+        except Exception as exc:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)